openupgradelib/openupgrade.py

```diff
@@ -799,29 +799,60 @@ def float_to_integer(cr, table, field):
 
 
 def map_values(
-        cr, source_column, target_column, mapping,
-        model=None, table=None, write='sql'):
+        cr, source_column, target_column, mapping, xml_mapping=False,
+        model=None, table=None, write='sql', check_completeness=False):
     """
     Map old values to new values within the same model or table. Old values
     presumably come from a legacy column.
     You will typically want to use it in post-migration scripts.
-
     :param cr: The database cursor
     :param source_column: the database column that contains old values to be \
     mapped
     :param target_column: the database column, or model field (if 'write' is \
     'orm') that the new values are written to
     :param mapping: list of tuples [(old value, new value)]
         Old value True represents "is set", False "is not set".
+    :param xml_mapping: If True, mapping is expected to be a fully qualified \
+    external ID as for example in "account.1_account_tax_1234".
     :param model: used for writing if 'write' is 'orm', or to retrieve the \
     table if 'table' is not given.
     :param table: the database table used to query the old values, and write \
     the new values (if 'write' is 'sql')
     :param write: Either 'orm' or 'sql'. Note that old ids are always \
     identified by an sql read.
+    :param check_completeness: True verifies, that all source entries \
+    have been accounted for in the mapping table and raises otherwise.
 
+    This moethod does not support mapping values across tables, neither \
+    relational nor property fields (it does not support resiliant \
+    handling of their respective relational tables - you still can use \
+    this method on those relational tables in sql mode).
     .. versionadded:: 8.0
     """
+    def use_xml_mapping(cr, mapping):
+        from openerp.modules.registry import RegistryManager
```

Comment:
Making this, you are restricting the use of this function on post_migration script. Maybe you can do it with plain SQL.